src/main/thrift/parquet.thrift

```diff
@@ -216,7 +216,12 @@ struct Statistics {
    /** count of distinct values occurring */
    4: optional i64 distinct_count;
    /**
-    * Min and max values for the column, determined by its ColumnOrder.
+    * lower and upper bound values for the column, determined by its ColumnOrder.
```

Comment:
IIRC, you want to allow truncation by adding a flag to indicate whether the min_value/max_value is truncated or not, right?

Reply:
Given the feature in https://issues.apache.org/jira/browse/PARQUET-1685, I want to assume that all existing stats are truncated. Going forward we should have a flag to explicitly indicate whether or not truncation took place and applications should perform aggregation pushdown only if that flag is found to indicate no truncation. But I think adding that flag can be tackled separately as a follow-up.

Reply:
I think they are relevant and can be discussed together.

What do you think? @gszadovszky @shangxinli

Reply:
Might not directly related to this question. If the column is a utf-8 column ( with LogicalType `UTF8` ), can the binary here be a non-utf8 bytes?

e.g. `s[...100]` is a valid utf-8, however, `s[...50]` is not.

Reply:
@mapleFU very good point, we should not allow truncation to produce a value that is invalid for given logical type
i think this is what @raunaqmorarka meant with `Such more compact values must still be valid values within the column's logical type.` in the code comment

Reply:
I agree, @wgtmac. I think, we should add the flag (if everyone agrees) along with the update of the min/max value spec.

Reply:
@wgtmac @gszadovszky could you please explain why the addition of a flag for truncation needs to happen together with the proposed change here ?
I believe https://issues.apache.org/jira/browse/PARQUET-1685 already broke the current spec by allowing for truncation of min/max for strings here. Even after we add a flag to indicate truncation, no application can safely assume that the min/max stats in existing parquet files are not truncated for strings. So I don't get why the addition of that flag shouldn't go as a separate change to the spec.

Reply:
I am fine going in separately but it needs to be released together. I think, the additional flag would complete fixing this situation. If an implementation uses min/max values as existing values and realize this spec change we would still give something to them to continue their approach. I agree if the flag is not there we cannot assume the min/max values are actual values.

Reply:
I've updated this PR to now include is_max_value_exact/is_min_value_exact for specifying when the max_value/min_value are the actual max and min values

Reply:
@wgtmac @gszadovszky can you please take a look at current changes ?

Reply:
@wgtmac @gszadovszky is anything more needed to merge this ?

Reply:
I just merged it. Thank you!

Reply:
Thanks!